statsmodels/tsa/statespace/regime_switching/kim_filter.py

```diff
@@ -0,0 +1,635 @@
+import numpy as np
+from scipy.misc import logsumexp
+from scipy.optimize import linprog
+from statsmodels.tsa.statespace.kalman_filter import KalmanFilter
+
+try:
+    from scipy.stats import multivariate_normal
+    multivariate_normal_logpdf = multivariate_normal.logpdf
+except ImportError:
+    def multivariate_normal_logpdf(x, mean=None, cov=None):
+        n = x.shape[0]
+        if np.linalg.matrix_rank(cov) < n:
+            raise RuntimeError('Cov matrix is singular.')
+        #TODO: to test this
+        return -0.5 * n * np.log(np.pi) - 0.5 * np.log(np.linalg.det(cov)) - \
```

Comment:
@josef-pkt What is the statsmodels policy on line continuations? I always use parentheses, but I can't remember if that's the policy or if it's just what I do.